packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -502,10 +507,46 @@ You've created a plugin project that doesn't yet support any platforms.
 ''');
 }
 
-void _printAddPlatformMessage(String pluginPath) {
+void _printPluginAddPlatformMessage(String pluginPath) {
   globals.printStatus('''
 To add platforms, run `flutter create -t plugin --platforms <platforms> .` under $pluginPath.
 For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''');
 }
+
+// shows warning if user requested a platform and if it's not enabled
+void _printWarningToEnablePlatforms(
+  List<String> requestedPlatforms,
+  [bool isApp = false]
+) {
+  final List<String> _showWarning = <String>[
+  if (requestedPlatforms.contains('android') && !featureFlags.isAndroidEnabled)
+    'android',
+  if (requestedPlatforms.contains('ios') && !featureFlags.isIOSEnabled)
+    'ios',
+  if (requestedPlatforms.contains('web') && !featureFlags.isWebEnabled)
+    'web',
+  if (requestedPlatforms.contains('macos') && !featureFlags.isMacOSEnabled)
+    'macos',
+  if (requestedPlatforms.contains('windows') && !featureFlags.isWindowsEnabled)
+    'windows',
+  if (requestedPlatforms.contains('linux') && !featureFlags.isLinuxEnabled)
+    'linux',
+  ];
+
+  if (_showWarning.isEmpty){
+    return;
+  }
+
+  final String _disabledPlatforms = _showWarning.join(', ');
+
+  globals.printStatus('! $_disabledPlatforms are not enabled; enable the platforms and then add them.',
```

Comment:
How about if platform is a desktop platform:
```
The desktop platforms: $_disabledPlatforms are not supported on your local environment. For more details, see: https://flutter.dev/desktop
```
And if the platform is web:
```
web is not currently supported on your local environment. For more details: see https://flutter.dev/docs/get-started/web
```

Reply:
This looks better. For Android and iOS, we just give them the command to enable them?

Reply:
I have made the changes, can you please look into the message for mobile (android, ios).